### DISCUSSION TITLE
Externalizar assets em projeto Angular 6

### DISCUSSION THREAD

**Discussion by raquelbazanelli** [frontendbr/forum#1382] [Perguntas e Respostas]
Bom dia pessoal, apesar de sempre ler o fórum, essa é minha primeira vez participando. 
Estou com problema em relação ao gerenciamento de assets. Atualmente tenho diversos projetos Angular 6 que possuem os mesmos estilos e imagens. Os assets estão repetidos nos projetos e a manutenção se tornou insustentável.
Estou procurando uma solução para externaliza-los. Pensei em colocar em um servidor separado e apontar o css no index.html, mas seria ruim quando estivesse trabalhando local, porque o Angular não deixa apontar para um arquivo que está na minha máquina.

Alguma sugestão?

**Comment by felipefialho** [comment]
Olá @raquelbazanelli, já pensou em deixar numa CDN? 

Pode ser uma alternativa.

**Comment by raquelbazanelli** [comment]
> Olá @raquelbazanelli, já pensou em deixar numa CDN?
> 
> Pode ser uma alternativa.

Já sim, mas no momento não tenho verba.

**Comment by adamfoerster** [comment]
@raquelbazanelli se entendi vc não consegue carregar arquivos externos quando está no localhost isso?
Você pode fazer esse css ser carregado condicionalmente a partir de um arquivo no projeto e a partir do o webapp estiver no ar  
`<script type="text/javascript">if (location.protocol != 'https:' && location.href.indexOf('localhost') == -1
) {...`
mais detalhes...
[https://stackoverflow.com/questions/15505225/inject-css-stylesheet-as-string-using-javascript](https://stackoverflow.com/questions/15505225/inject-css-stylesheet-as-string-using-javascript)

**Comment by raquelbazanelli** [comment]
> @raquelbazanelli se entendi vc não consegue carregar arquivos externos quando está no localhost isso?
> Você pode fazer esse css ser carregado condicionalmente a partir de um arquivo no projeto e a partir do o webapp estiver no ar
> `<script type="text/javascript">if (location.protocol != 'https:' && location.href.indexOf('localhost') == -1 ) {...`
> mais detalhes...
> https://stackoverflow.com/questions/15505225/inject-css-stylesheet-as-string-using-javascript

Interessante, não conhecia, não é exatamente isso que eu precisava, mas essa parte eu consegui eliminar.
Provavelmente vou seguir nessa linha de importar o css no index.html, posso estar enganada mas não me parece ser a melhor solução. :(

**Comment by felipefialho** [comment]
@raquelbazanelli Pensei aqui, outra saída seria automatizar a cópia desses assets da pasta principal toda vez que você rodar ou der build no projeto.

Essa automatização poderia ser feita até mesmo usando npm scripts.
